From e0947b01bd636dfbb59f87f0008eb8cdbc26e845 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jonas=20=C3=85dahl?= <jadahl@gmail.com>
Date: Wed, 18 Sep 2019 14:57:48 +0200
Subject: [PATCH 127/152] introspect: Check whitelist also for
 GetRunningWindows()

Otherwise the xdg-desktop-portal-gtk screen cast widget won't work.

https://gitlab.gnome.org/GNOME/gnome-shell/merge_requests/732
---
 js/misc/introspect.js | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/js/misc/introspect.js b/js/misc/introspect.js
index c7fb3a712..f256555cb 100644
--- a/js/misc/introspect.js
+++ b/js/misc/introspect.js
@@ -127,7 +127,8 @@ var IntrospectService = class {
         let apps = this._appSystem.get_running();
         let windowsList = {};
 
-        if (!this._isIntrospectEnabled()) {
+        if (!this._isIntrospectEnabled() &&
+            !this._isSenderWhitelisted(invocation.get_sender())) {
             invocation.return_error_literal(Gio.DBusError,
                                             Gio.DBusError.ACCESS_DENIED,
                                             'App introspection not allowed');
-- 
2.23.0

